cmd/metal-api/internal/service/switch-service.go

```diff
@@ -77,6 +77,16 @@ func (r switchResource) webService() *restful.WebService {
 		Returns(http.StatusCreated, "Created", v1.SwitchResponse{}).
 		DefaultReturns("Error", httperrors.HTTPErrorResponse{}))
 
+	ws.Route(ws.POST("/").
+		To(admin(r.updateSwitch)).
+		Operation("updateSwitch").
+		Doc("updates a switch. if the switch was changed since this one was read, a conflict is returned").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Reads(v1.SwitchUpdateRequest{}).
+		Returns(http.StatusOK, "OK", v1.SwitchResponse{}).
+		Returns(http.StatusConflict, "Conflict", httperrors.HTTPErrorResponse{}).
```

Comment:
The `updateSwitch` function does not return a conflict status code explicitly, I think this line can be removed. Should be covered by default response